dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -552,6 +563,25 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
         return tokenDAO.findByUserId(user.getId());
     }
 
+    /**
+     * Construct a URL for Dockstore
+     * @param dockstoreScheme Dockstore scheme, e.g. http or https
+     * @param dockstoreHostName Dockstore host name, e.g. dockstore.org
+     * @param dockstorePort Dockstore port, e.g. 4200
+     */
+    private String createHostUrl(String dockstoreScheme, String dockstoreHostName, String dockstorePort) {
```

Comment:
There should be existing code for this kind of thing. 
Perhaps https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#URL(java.lang.String,%20java.lang.String,%20int,%20java.lang.String,%20java.net.URLStreamHandler) and then toString or https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/client/utils/URIBuilder.html

Reply:
done